Module not found: TypeError: Cannot read property 'request' of null

# Bug report

I get the following error when running the newest version of webpack `v4.28.1`.

```
Module not found: TypeError: Cannot read property 'request' of null
```

It happens whenever I try and import a dependency that only works on Node.JS that I've set to an empty object (for example `fs`).

Here is my webpack externals config.

```js
webpackConfig.externals = {
  child_process: '{}',
  deasync: '{}',
  fs: '{}',
  winston: '{}'
}
```

Please note that this works fine on webpack `v4.27.1`.

Here is the full stack trace.

```
ERROR in ./node_modules/@reactant/core/lib/log/winston.js
Module not found: TypeError: Cannot read property 'request' of null
 @ ./node_modules/@reactant/core/lib/log/winston.js 17:15-33
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in ./node_modules/fs-extra/lib/index.js
Module not found: TypeError: Cannot read property 'request' of null
 @ ./node_modules/fs-extra/lib/index.js 23:11-24
 @ ./node_modules/@reactant/core/lib/log/winston.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in ./node_modules/fs-extra/lib/empty/index.js
Module not found: TypeError: Cannot read property 'request' of null
 @ ./node_modules/fs-extra/lib/empty/index.js 4:11-24
 @ ./node_modules/fs-extra/lib/index.js
 @ ./node_modules/@reactant/core/lib/log/winston.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/getOS/index.js
Module not found: TypeError: Cannot read property 'request' of null
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/getOS/index.js 90:21-45
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/os.js
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/index.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/getOS/index.js
Module not found: TypeError: Cannot read property 'request' of null
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/getOS/index.js 89:16-34
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/os.js
 @ /Users/codejamninja/Projects/reactant/packages/cli/node_modules/js-info/lib/index.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in ./node_modules/graceful-fs/graceful-fs.js
Module not found: TypeError: Cannot read property 'request' of null
 @ ./node_modules/graceful-fs/graceful-fs.js 1:9-22
 @ ./node_modules/fs-extra/lib/move-sync/index.js
 @ ./node_modules/fs-extra/lib/index.js
 @ ./node_modules/@reactant/core/lib/log/winston.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js

ERROR in ./node_modules/jsonfile/index.js
Module not found: TypeError: Cannot read property 'request' of null
 @ ./node_modules/jsonfile/index.js 5:8-21
 @ ./node_modules/fs-extra/lib/json/jsonfile.js
 @ ./node_modules/fs-extra/lib/json/index.js
 @ ./node_modules/fs-extra/lib/index.js
 @ ./node_modules/@reactant/core/lib/log/winston.js
 @ ./node_modules/@reactant/core/lib/log/index.js
 @ ./node_modules/@reactant/core/lib/index.js
 @ /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
 @ multi ./web/client.js /Users/codejamninja/Projects/reactant/packages/web-isomorphic/lib/hotDevClient.js
```

Instead of throwing an error, it should simply require the module as an empty object.
Because the error is happening inside of webpack, the stack trace does not tell me what is going wrong. I've searched all over for this supposed 'test' property. I am not using a property called test anywhere in my code.

This also fails with other node-specific npm modules, such as `winston` and `deasync`.

**Other relevant information:**
webpack version:  
Node.js version: v8.14.0
Also tested on Node.js version: v10.14.
Operating System: Ubuntu 18.10

--- a/lib/IgnorePlugin.js
+++ b/lib/IgnorePlugin.js
@@ -38,6 +38,8 @@ class IgnorePlugin {
 	 * @returns {TODO|null} returns result or null if result should be ignored
 	 */
 	checkIgnore(result) {
+		if (!result) return result;
+
 		if (
 			"checkResource" in this.options &&
 			this.options.checkResource &&

--- a/test/configCases/ignore/multiple-with-externals/ignored-module1.js
+++ b/test/configCases/ignore/multiple-with-externals/ignored-module1.js
@@ -1,0 +1,1 @@
+module.exports = "ignored";

--- a/test/configCases/ignore/multiple-with-externals/ignored-module2.js
+++ b/test/configCases/ignore/multiple-with-externals/ignored-module2.js
@@ -1,0 +1,1 @@
+module.exports = "ignored";

--- a/test/configCases/ignore/multiple-with-externals/normal-module.js
+++ b/test/configCases/ignore/multiple-with-externals/normal-module.js
@@ -1,0 +1,1 @@
+module.exports = "normal";

--- a/test/configCases/ignore/multiple-with-externals/test.js
+++ b/test/configCases/ignore/multiple-with-externals/test.js
@@ -1,0 +1,18 @@
+/* globals it */
+"use strict";
+
+it("should ignore ignored resources 1", function() {
+	expect(function() {
+		require("./ignored-module1");
+	}).toThrowError();
+});
+it("should ignore ignored resources 2", function() {
+	expect(function() {
+		require("./ignored-module2");
+	}).toThrowError();
+});
+it("should not ignore resources that do not match", function() {
+	expect(function() {
+		require("./normal-module");
+	}).not.toThrowError();
+});

--- a/test/configCases/ignore/multiple-with-externals/webpack.config.js
+++ b/test/configCases/ignore/multiple-with-externals/webpack.config.js
@@ -1,0 +1,18 @@
+"use strict";
+
+const IgnorePlugin = require("../../../../lib/IgnorePlugin");
+
+module.exports = {
+	entry: "./test.js",
+	externals: {
+		"./normal-module": "{}"
+	},
+	plugins: [
+		new IgnorePlugin({
+			resourceRegExp: /ignored-module1/
+		}),
+		new IgnorePlugin({
+			resourceRegExp: /ignored-module2/
+		})
+	]
+};

